requirements.txt

```diff
@@ -0,0 +1,15 @@
+bcolz
+h5py
+matplotlib
+moto
+pandas_datareader
+partd
+pytest
```

Comment:
Also what about optional dependencies of other projects that users might run into, pytables for dask.dataframe (implicitly depended on from pandas)

Reply:
Personally don't want to go down this road. It's too hard to tell what those might be and this goes far afield fast.

Reply:
I agree that it becomes a slippery slope, however I also suspect that pytables is more likely to be an issue for users than is scipy.

If the goal is to avoid packaging issues then it probably makes sense to also raise issues throughout the community?  It seems odd to be the only ones doing this.

Reply:
Sorry I disagree. As we are not using `pytables` directly, it should not be a dependency. If we are concerned about how `dask` affects `pytables`, we should consider testing against it. At that point it becomes a testing dependency and can be listed for that reason.

To put this more generally, the reasoning should not be users may want to use this with Dask so it should be a dependency. Instead the reasoning should be, we aim to be compatible with this useful library through our testing and signify this already present commitment by listing it as a testing dependency. IOW dependency comes after usage not before.

Reply:
> If the goal is to avoid packaging issues then it probably makes sense to also raise issues throughout the community? It seems odd to be the only ones doing this.

This happens often through engagement with packaging ecosystems like conda-forge and  Linux distros. We also encourage projects to package licenses, changelogs, and point out other issues in a libraries metadata or build systems. In many cases `CMakeLists.txt` have been written for many projects by Windows users'/maintainers' in conda-forge seeking Windows support in libraries that had none. IOW this sort of stuff does happen.